### DISCUSSION TITLE
Facing lots of issues while following Contribution-quick start guidelines for M1.

### DISCUSSION THREAD

**Discussion by nikhilkarve** [apache/airflow#42637] [General]
### What do you see as an issue?

1. While running breeze ci-image build
ERROR: failed to solve: process "/bin/bash -o pipefail -o errexit -o nounset -o nolog -c bash /scripts/docker/install_os_dependencies.sh dev" did not complete successfully: exit code: 100
Error when building image! Image build: 3.8:linux/arm64
2. After running breeze setup autocomplete
Users/nkarve/Desktop/Personal/airflow/airflow/dev/breeze/autocomplete/breeze-complete-zsh.sh:34: command not found: compdef

Also had a few errors earlier with docker versions. 

### Solving the problem

It feels overwhelming and difficult for a new person to get started. Can we please have a dedicated macOs dev env installation guide? Or at least a youtube video from someone who have done it before. 

Much needed. Thanks!

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by gopidesupavan** [comment]
It looks like you have problem in bash or zshrc file, try add these to those files.

autoload -Uz compinit
compinit

**Comment by potiuk** [comment]
> It feels overwhelming and difficult for a new person to get started. Can we please have a dedicated macOs dev env installation guide? Or at least a youtube video from someone who have done it before.

Yes. Many people have M1 macbook and it works for them following the instructions. If you can help to identify the problems in your particular setup (@nikhilkarve ) that cause it, it would be great if you find out and add some troubleshooting documentation for others who might have similar problem, but it's impossible to know what problem you have other than guessing (because it is not reproducible).

My best guess is that you migrated your environment from pre-M1 Mac (intel) to the M1 Mac and it broke architecture support for your system (so basically you have intel binaries of Python installed  but no emulation enabled). Looks like your OS + Python + Docker installation is broken because of that migration and the best you can do is to wipe it all out and install everything from the scratch using M1 architecture.

**Comment by potiuk** [comment]
Converted it to a discussion as this seems to be isolated issue of a user with broken environment,
